File: Data Structures and Algorithms/Algorithms/java/linearSearch.java
package college;

public class linearSearch {

	public static void main(String[] args) {
		int ar[] = {1,2,5,8,12,27, 31,42};
		int element = 27;
		int ans = linearSearch(ar, 0, 7, element);
		if (ans!=-1)
				System.out.println("element found at "+ans);
		else
			System.out.println("not found");
	}
	static int linearSearch(int [] ar, int start, int end, int element) {
        if (end < start) 
           return -1; 
        if (ar[start] == element) 
           return start; 
        if (ar[end] == element) 
           return end; 
        return linearSearch(ar, start+1, end-1, element); 
	}
}

File: Data Structures and Algorithms/Algorithms/python/linearSearch.py
# Program to find an element using linear search


def linear_search(arr, x):

    for i in range(len(arr)):

        if arr[i] == x:
            return i

    return -1


arr = [2, 1, 3, 5, 6, 7]
x = 3
print(f'Position of element {x} is at {linear_search(arr, x)}')
